File: scode/NullByte00/prob17/main.cpp
#include <iostream>
#include <set>
#include <stack>

int main() {
    int sum=0;
    std::string s;
    std::cin >> s;

    std::set<std::string> possibilities;
    std::stack<std::string> stack;
    stack.push(s);

    while (!stack.empty()) {
        // std::cout << "before: " << stack.top() << " " << possibilities.size() << std::endl;

        s = stack.top();
        stack.pop();

        if (possibilities.find(s) != possibilities.end()) {
            continue;
        } else {
            possibilities.insert(s);
            sum++;
        }

        if (s.length() > 1) {
            for (int i = 0; i < s.length() - 1; ++i) {
                if (s[i] == s[i + 1]) {
                    if (s[i] == 't') {
                        std::string s1 = s.substr(0, i) + "v" + s.substr(i + 2);
                        stack.push(s1);
                    } else {
                        stack.push(s.substr(0, i) + "t" + s.substr(i + 2));
                    }
                }
            }
        }
        // std::cout << "after: " << stack.size() << " " << possibilities.size() << std::endl;
    }

    //std::cout << possibilities.size() << std::endl;
    std::cout << sum;
    return 0;
}

File: scode/NullByte00/prob17/Main.java
import java.util.HashSet;
import java.util.Stack;

public class Main {
    public static void main(String[] args) {
        java.util.Scanner scanner = new java.util.Scanner(System.in);
        String s = scanner.nextLine();

        HashSet<String> possibilities = new HashSet<>();
        Stack<String> stack = new Stack<>();
        stack.push(s);

        while (!stack.isEmpty()) {
            // System.out.println("before: " + stack.peek() + " " + possibilities.size());

            s = stack.pop();

            if (possibilities.contains(s)) {
                continue;
            } else {
                possibilities.add(s);
            }

            if (s.length() > 1) {
                for (int i = 0; i < s.length() - 1; ++i) {
                    if (s.charAt(i) == s.charAt(i + 1)) {
                        if (s.charAt(i) == 't') {
                            String s1 = s.substring(0, i) + "v" + s.substring(i + 2);
                            stack.push(s1);
                        } else {
                            stack.push(s.substring(0, i) + "t" + s.substring(i + 2));
                        }
                    }
                }
            }
            // System.out.println("after: " + stack.size() + " " + possibilities.size());
        }
    }
}